#594 (1) -- consolidate configs and prefs

CarlosNZ opened the pull request:
First part of #594, as outlined in [this comment](https://github.com/openmsupply/application-manager-server/issues/594#issuecomment-928235451)

Requires [front-end #1011](https://github.com/openmsupply/application-manager-web-app/pull/1011)

nmadruga commented on src/components/permissions/routes.ts:
```diff
@@ -116,11 +119,23 @@ const routeVerification = async (request: any, reply: any) => {
+  const prefs = JSON.parse(
+    readFileSync(path.join(getAppEntryPointDir(), '../preferences.json'), 'utf8')
+  )
+  const languageOptions = JSON.parse(
+    readFileSync(path.join(getAppEntryPointDir(), '../localisation/languages.json'), 'utf8')
+  )
+  reply.send({ preferences: prefs.web, languageOptions })
```
Hmm I see so this would get passed in the first iteration between server and front-end app?

CarlosNZ replied:
Yeah, so the front-end starts off by loading prefs and languageOptions before it does anything else.

It already did that for languageOptions, I've just updated it so it gets prefs at the same time.

nmadruga replied:
Cool

CarlosNZ commented:
Oh, one thing I forgot to mention @nmadruga -- I've moved the JWT secret out of the config file and moved it into the `.env` file along with SMTP password and WEB_HOST (so it's not shared in the repo and elsewhere).

We'll probably need to update the build script/instructions a little bit.

nmadruga commented:
> Oh, one thing I forgot to mention @nmadruga -- I've moved the JWT secret out of the config file and moved it into the `.env` file along with SMTP password and WEB_HOST (so it's not shared in the repo and elsewhere).
> 
> We'll probably need to update the build script/instructions a little bit.

Ok - that's good. Have you added in an entry on Bitwarden? I think we need to add this to our docs straight away - so don't need to deal with that on next time we just need to release a version.

CarlosNZ commented:
> Ok - that's good. Have you added in an entry on Bitwarden? I think we need to add this to our docs straight away - so don't need to deal with that on next time we just need to release a version.

We don't need any particular key to be saved. We can just make one up each time we restart the server (which will mean current JWTs will no longer be valid).

nmadruga commented on preferences.json:
```diff
@@ -0,0 +1,13 @@
+  },
+  "web": {
+    "paginationPresets": [2, 5, 10, 20, 50],
+    "defaultLanguageCode": "en_nz"
```
Ok, the default is defined here... I think this answers one question I had while reviewing the Front-end PR. 😄

nmadruga commented:
One thing that I noticed, not sure if it is expected behavior - I changed to a different language before login and then selected on user to login in (admin) after I logged in the language changed back to English. Shouldn't it stayed with my selected one?

CarlosNZ commented:
> One thing that I noticed, not sure if it is expected behavior - I changed to a different language before login and then selected on user to login in (admin) after I logged in the language changed back to English. Shouldn't it stayed with my selected one?

Currently, when logging out, the local storage gets completely wiped, so the login screen will go back to the default language.

I can make it so that it only deletes the JWT and leaves the language preference untouched. I was slightly reluctant to do that, since in the past we've had problems with Apollo cache stuff hanging around in local storage and messing things up, so it was safer to fully delete the cache.

What I might do is delete the cache as it does, but then immediate re-write just the language setting back. Effectively -- delete everything BUT the language setting. That sound okay?

CarlosNZ commented:
> What I might do is delete the cache as it does, but then immediate re-write just the language setting back. Effectively -- delete everything BUT the language setting. That sound okay?

Okay, done that: 33df78ec
(I also created a tiny `/logout` route which I've been wanting to do for ages. Sometimes the app get itself stuck in some weird state with nothing on screen, so can just manually go to /logout url to reset. )

nmadruga reviewed (approved):
Yes, I was a bit confused with the change on languages before. And thanks for adding the strings to core_templates, that will be useful to have there.
